src/_tutorials/dart-vm/httpserver.md

```diff
@@ -555,66 +560,55 @@ The `close()` method sends the request to the server
-       'BFF':      'Chewbacca',
-       'ship':     'Millennium Falcon',
-       'weakness': 'smuggling debts'
-     };
-
-[[note]]1[[/note]]  [[highlight]]var request = await[[/highlight]] new HttpClient().post(
-[[note]]2[[/note]]      [[highlight]]InternetAddress.LOOPBACK_IP_V4.host, 4049, '/file.txt');[[/highlight]]
-[[note]]3[[/note]]  [[highlight]]request.headers.contentType = ContentType.JSON;[[/highlight]]
-[[note]]4[[/note]]  [[highlight]]request.write(JSON.encode(jsonData));[[/highlight]]
-[[note]]5[[/note]]  [[highlight]]HttpClientResponse response = await request.close()[[/highlight]];
-[[note]]6[[/note]]  await for (var contents in response.transform([[highlight]]UTF8.decoder[[/highlight]])) {
-       print(contents);
-    }
+import 'dart:async';
+import 'dart:io';
+import 'dart:convert' show UTF8, JSON;
+
+Future main() async {
+  Map jsonData = {
+    'name': 'Han Solo',
+    'job': 'reluctant hero',
+    'BFF': 'Chewbacca',
+    'ship': 'Millennium Falcon',
+    'weakness': 'smuggling debts'
+  };
+
+  var request = await new HttpClient()
+      [!/*1*/!] .post(InternetAddress.LOOPBACK_IP_V4.host, 4049, 'file.txt');
+  [!/*2*/!] request.headers.contentType = ContentType.JSON;
+  [!/*3*/!] request.write(JSON.encode(jsonData));
+  [!/*4*/!] HttpClientResponse response = await request.close();
+  [!/*5*/!] await for (var contents in response.transform([!UTF8.decoder!])) {
+    print(contents);
   }
+}
 {% endprettify %}
 <div class="prettify-filename">basic_writer_client.dart</div>
 
-<span class="code-note">1</span>
-When the `post()` connection succeeds (and execution resumes after
-the await expression), the returned [HttpClientRequest][]
-object is assigned to the `request` variable.
-
-<span class="code-note">2</span>
-The `post()` method requires the host, port, and the path to the requested
-resource.
-In addition to `post()`, the HttpClient class provides functions
-for making other kinds of
-requests, including `postUrl()`, `get()`, and `open()`.
-
-<span class="code-note">3</span>
-An HttpClientRequest object has an HttpHeaders object,
-which contains the request headers.
-For some headers,
-like `contentType`,
-HttpHeaders has a property specific to that header.
-For other headers, use the `set()` method to
-put the header in the HttpHeaders object.
-
-<span class="code-note">4</span>
-The client writes data to the request object using `write()`.
-The encoding, JSON in this example,
-matches the type specified in the ContentType header.
-
-<span class="code-note">5</span>
-The `close()` method sends the request to the server and, when complete,
-returns an [HttpClientResponse][] object
-that's assigned to the `reponse` variable.
-
-<span class="code-note">6</span>
-The response from the server is encoded in UTF-8.
-Use a transformer defined in the `dart:convert` library
-to convert the data into regular Dart string format.
+{:.code-notes}
+1. The `post()` method requires the host, port, and the path to the requested
```

Comment:
Why did you remove the discussion of `request` and `await`?

Reply:
For the same reasons that I removed a similar line earlier in the page: see [this comment](https://github.com/dart-lang/site-www/pull/463/files#r155930657):

1. The code is self-explanatory.
2. We've already explained what `await` does.
3. I don't think that we need to describe how a variable assignment occurs.

Let me know if you'd like for me to add it back.

Reply:
I think the main reason to keep it would be to emphasize that the returned value is a HttpClientRequest, since that isn't visible in the code. (You mention that type later, but you don't explicitly tie it to this variable.)

Reply:
Yes, good point. I've added a type annotation to the local request variable. Note that I'm not setting a precedent here:

- A little earlier in the page, in the first `number_thinker.dart` excerpt of the "Listening for and handling requests" section, the local variable `server` has a type annotation.
- The `number_guesser.dart` client also has its local `request` variable typed.